components/automate-deployment/pkg/deployment/deployment.go

```diff
@@ -344,7 +366,7 @@ func (d *Deployment) UpdateExpectedServicesFromManifest() error {
 		}
 	}
 
-	// Remove any removed services
+	// Update the expected services with our new ExpectedServices
```

Comment:
Seeing this loop, I'm a bit confused about why the above deletion loop is needed.  Because the range here is over the serviceIDs, the new expectedServices array should be free from any removed services without the deletion loop we added above.

Reply:
Yeah, you're right. Not sure what I was doing here.